src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -15296,177 +15282,182 @@ namespace SurfaceGeometry {
-            Y(n) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(n).y;
-            Z(n) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(n).z;
+        auto &surfaceTmp = state.dataSurfaceGeometry->SurfaceTmp(SurfNum);
+        auto &vertices = surfaceTmp.Vertex;
+
+        for (int n = 1; n <= NSides; ++n) {
+            X(n) = vertices(n).x;
+            Y(n) = vertices(n).y;
+            Z(n) = vertices(n).z;
         }
-        X(NSides + 1) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(1).x;
-        Y(NSides + 1) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(1).y;
-        Z(NSides + 1) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(1).z;
-        X(NSides + 2) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(2).x;
-        Y(NSides + 2) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(2).y;
-        Z(NSides + 2) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(2).z;
+        X(NSides + 1) = vertices(1).x;
+        Y(NSides + 1) = vertices(1).y;
+        Z(NSides + 1) = vertices(1).z;
+        X(NSides + 2) = vertices(2).x;
+        Y(NSides + 2) = vertices(2).y;
+        Z(NSides + 2) = vertices(2).z;
 
         // Determine a suitable plane in which to do the tests
-        Det = 0.0;
-        for (n = 1; n <= NSides; ++n) {
+        Real64 Det = 0.0;
+        for (int n = 1; n <= NSides; ++n) {
             Det += X(n) * Y(n + 1) - X(n + 1) * Y(n);
         }
         if (std::abs(Det) > 1.e-4) {
             A = X;
             B = Y;
         } else {
             Det = 0.0;
-            for (n = 1; n <= NSides; ++n) {
+            for (int n = 1; n <= NSides; ++n) {
                 Det += X(n) * Z(n + 1) - X(n + 1) * Z(n);
             }
             if (std::abs(Det) > 1.e-4) {
                 A = X;
                 B = Z;
             } else {
                 Det = 0.0;
-                for (n = 1; n <= NSides; ++n) {
+                for (int n = 1; n <= NSides; ++n) {
                     Det += Y(n) * Z(n + 1) - Y(n + 1) * Z(n);
                 }
                 if (std::abs(Det) > 1.e-4) {
                     A = Y;
                     B = Z;
                 } else {
                     // This condition should not be reached if the surfaces are guaranteed to be planar already
-                    ShowSevereError(state, "CheckConvexity: Surface=\"" + state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Name + "\" is non-planar.");
+                    ShowSevereError(state, "CheckConvexity: Surface=\"" + surfaceTmp.Name + "\" is non-planar.");
                     ShowContinueError(state, "Coincident Vertices will be removed as possible.");
-                    for (n = 1; n <= state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides; ++n) {
-                        auto const &point(state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(n));
+                    for (int n = 1; n <= surfaceTmp.Sides; ++n) {
```

Comment:
The rest is a bunch of cleanup / modernizing. Cf b929068cb2d2276eb3759955a921150fded71414 for the commit in which I just added a few references to shorten the code and improve readability